kotsadm/web/src/components/snapshots/Snapshots.jsx

```diff
@@ -1,131 +1,285 @@
+
+  startInstanceSnapshot = () => {
+    const fakeProgressSnapshot = {
+      name: "Preparing snapshot",
+      status: "InProgress",
+      trigger: "manual",
+      sequence: "",
+      startedAt: moment().format("MM/DD/YY @ hh:mm a"),
+      finishedAt: "",
+      expiresAt: "",
+      volumeCount: 0,
+      volumeSuccessCount: 0,
+      volumeBytes: 0,
+      volumeSizeHuman: ""
+    }
+
+    this.setState({
+      startingSnapshot: true,
+      startSnapshotErr: false,
+      startSnapshotErrorMsg: "",
+      isStartButtonClicked: true,
+      snapshots: [...this.state.snapshots, fakeProgressSnapshot].sort((a, b) => new Date(b.startedAt) - new Date(a.startedAt))
     });
 
-    fetch(`${window.env.API_ENDPOINT}/snapshots/settings`, {
-      method: "GET",
+    fetch(`${window.env.API_ENDPOINT}/snapshot/backup`, {
+      method: "POST",
       headers: {
         "Authorization": Utilities.getToken(),
         "Content-Type": "application/json",
       }
     })
-      .then(res => res.json())
-      .then(result => {
-        this.setState({
-          snapshotSettings: result,
-          isLoadingSnapshotSettings: false,
-          snapshotSettingsErr: false,
-          snapshotSettingsErrMsg: "",
-        })
-        if (result.veleroVersion === "") {
-          setTimeout(() => {
-            this.setState({ hideCheckVeleroButton: false });
-          }, 5000);
+      .then(async (result) => {
+        if (result.ok) {
+          this.setState({
+            startingSnapshot: false
+          });
         } else {
-          this.setState({ hideCheckVeleroButton: false });
+          const body = await result.json();
+          this.setState({
+            startingSnapshot: false,
+            startSnapshotErr: true,
+            startSnapshotErrorMsg: body.error,
+            snapshots: this.state.snapshots.filter(s => s.name !== "Preparing snapshot")
+          });
         }
       })
       .catch(err => {
         this.setState({
-          isLoadingSnapshotSettings: false,
-          snapshotSettingsErr: true,
-          snapshotSettingsErrMsg: err,
+          startingSnapshot: false,
+          startSnapshotErr: true,
+          startSnapshotErrorMsg: err,
+          snapshots: this.state.snapshots.filter(s => s.name !== "Preparing snapshot")
         })
       })
   }
 
-  componentDidMount = () => {
-    this.fetchSnapshotSettings();
-  }
+  toggleConfirmDeleteModal = snapshot => {
+    if (this.state.deleteSnapshotModal) {
+      this.setState({ deleteSnapshotModal: false, snapshotToDelete: "", deleteErr: false, deleteErrorMsg: "" });
+    } else {
+      this.setState({ deleteSnapshotModal: true, snapshotToDelete: snapshot, deleteErr: false, deleteErrorMsg: "" });
+    }
+  };
 
-  toggleSnapshotView = (isEmptyView) => {
-    this.setState({ toggleSnapshotView: !this.state.toggleSnapshotView, isEmptyView: isEmptyView ? isEmptyView : false });
-  }
+  handleDeleteSnapshot = snapshot => {
+    const fakeDeletionSnapshot = {
```

Comment:
Why do we have a fake delete?

Reply:
To show deleting row because we're pulling snapshot data and it can be that it's not there immediately

Reply:
Ah gotcha